extensions/wikia/Blogs/monobook/SpecialCreateBlogPage.php

```diff
@@ -153,99 +155,102 @@ protected function save() {
 
 			default:
 				Wikia::log( __METHOD__, "editpage", $status->value );
-				if( $status->value == EditPage::AS_READ_ONLY_PAGE_LOGGED ) {
-					$sMsg = wfMsg('create-blog-cant-edit');
+				if ( $status->value == EditPage::AS_READ_ONLY_PAGE_LOGGED ) {
+					$sMsg = wfMsg( 'create-blog-cant-edit' );
 				}
 				else {
-					$sMsg = wfMsg('create-blog-spam');
+					$sMsg = wfMsg( 'create-blog-spam' );
 				}
 				$this->mFormErrors[] = $sMsg . '(' . $status->value . ')';
 				$this->renderForm();
 				break;
 		}
-
 	}
 
 	protected function parseFormData() {
-		global $wgUser, $wgRequest, $wgOut;
-
-		wfRunHooks('BlogsAlternateEdit', array(false));
-
-		$this->mFormData['postId'] = $wgRequest->getVal('blogPostId');
-		$this->mFormData['postTitle'] = $wgRequest->getVal('blogPostTitle');
-		$this->mFormData['postBody'] = $wgRequest->getVal('wpTextbox1');
-		$this->mFormData['postEditSummary'] = $wgRequest->getVal('wpSummary');
-		$this->mFormData['postCategories'] = $wgRequest->getVal('wpCategoryTextarea1');
-		$this->mFormData['isVotingEnabled'] = $wgRequest->getCheck('blogPostIsVotingEnabled');
-		$this->mFormData['isCommentingEnabled'] = $wgRequest->getCheck('blogPostIsCommentingEnabled');
-		$this->mFormData['isExistingArticleEditAllowed'] = $wgRequest->getVal('articleEditAllowed');
+		global $wgUser, $wgRequest;
+
+		$token = $wgRequest->getVal( 'wpEditToken' );
+		if ( !$wgUser->matchEditToken( $token ) ) {
+			$this->mFormErrors[] = wfMessage( 'create-blog-cant-edit' )->text();
```

Comment:
@garthwebb what do you think about more descriptive error message? Maybe ```sessionfailure```?

BTW: message should be escaped.